wp-content/civi-extensions/goonjcustom/api/v3/Goonjcustom/DroppingCenterOutcomeCron.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+/**
+ * @file
+ */
+
+use Civi\Api4\Activity;
+use Civi\Api4\EckEntity;
+
+/**
+ * Cron job to update Dropping Center outcomes.
+ */
+function civicrm_api3_goonjcustom_dropping_center_outcome_cron($params) {
+  $returnValues = [];
+
+  $cashContributionById = [];
+  $productSaleAmountById = [];
+
+  $droppingCenterMetas = EckEntity::get('Dropping_Center_Meta', TRUE)
+    ->addSelect('Donation.Cash_Contribution', 'Donation.Product_Sale_Amount_GBG_', 'Dropping_Center_Meta.Dropping_Center')
+    ->addClause('OR', ['Donation.Cash_Contribution', 'IS NOT EMPTY'], ['Donation.Product_Sale_Amount_GBG_', 'IS NOT EMPTY'])
+    ->execute();
+
+  foreach ($droppingCenterMetas as $center) {
+    $id = $center['Dropping_Center_Meta.Dropping_Center'];
+    $cashContributionById[$id] = ($cashContributionById[$id] ?? 0) + (float) ($center['Donation.Cash_Contribution'] ?? 0);
+    $productSaleAmountById[$id] = ($productSaleAmountById[$id] ?? 0) + (float) ($center['Donation.Product_Sale_Amount_GBG_'] ?? 0);
+  }
+
+  $vehicleDispatches = EckEntity::get('Collection_Source_Vehicle_Dispatch', TRUE)
+    ->addSelect('Camp_Vehicle_Dispatch.Collection_Camp')
+    ->addWhere('Camp_Vehicle_Dispatch.Collection_Camp', 'IS NOT NULL')
+    ->execute();
+
+  $vehicleDispatchArray = $vehicleDispatches->getIterator()->getArrayCopy();
```

Comment:
What we are trying to do here.

Reply:
Converting that data into an array, and counting how many dispatch records exist for vehicle dispatch